### PR TITLE
WIP: Remove managers from metadataonly

### PR DESCRIPTION
**Pull request by apelisse**
meta: Remove ManagedFields from metadataonly responses
    
ManagerFields can be a significantly big field and is certainly not
needed by most metadata-only clients right now, especially since they
probably use the metadata-only API because they are looking for smaller
objects.
    
Just remove the field from the response so that they go back to a
regular size.

**What type of PR is this?**
/kind failing-test

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/apimachinery/pkg/api/meta/meta.go [hunk 1]
```diff
@@ -135,7 +137,10 @@ func AsPartialObjectMetadata(m metav1.Object) *metav1.PartialObjectMetadata {
 				OwnerReferences:            m.GetOwnerReferences(),
 				Finalizers:                 m.GetFinalizers(),
 				ClusterName:                m.GetClusterName(),
-				ManagedFields:              m.GetManagedFields(),
+				// Explicitely set ManagedFields to nil
+				// since it's big and not necessary for
+				// metadata-only clients.
```
Comment (lavalamp) [inline staging/src/k8s.io/apimachinery/pkg/api/meta/meta.go]: // TODO: we may need to pass this through if requested by the user. But this isn't particularly useful without the entire object, so maybe not.

Comment (apelisse) [reply staging/src/k8s.io/apimachinery/pkg/api/meta/meta.go]: Yeah, I thought about that, but "not passing" is probably a reasonable default, so we can provide a mechanism later.

Comment (lavalamp) [reply staging/src/k8s.io/apimachinery/pkg/api/meta/meta.go]: That's my thought too, I just think we should probably say so in the comment.

Comment (apelisse) [reply staging/src/k8s.io/apimachinery/pkg/api/meta/meta.go]: I'll update once we're done with the first round of tests.

**Comment by apelisse** [comment]
@liggitt I'm not sure what the errors are for the gc, but clearly doesn't look like one.

**Comment by apelisse** [comment]
> `[sig-storage] In-tree Volumes [Driver: local][LocalVolumeType: dir-link-bindmounted] [Testpattern: Pre-provisioned PV (default fs)] subPath should support readOnly directory specified in the volumeMount`

I don't think this is GC either.

**Comment by liggitt** [comment]
50% of runs so far failed with GC or quota timeouts

**Comment by apelisse** [comment]
```
> curl -s https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/88098/pull-kubernetes-e2e-gce/1227820303124533253/artifacts/e2e-6cdf6313b7-674b9-master/kube-controller-manager.log | grep -a garbagecollector | grep gc-9825 | less
I0213 05:31:03.267455       1 garbagecollector.go:404] processing item [v1/Pod, namespace: gc-9825, name: pod3, uid: 3e113e73-96f7-41aa-a0f0-2c45e53413ed]
I0213 05:31:03.271057       1 garbagecollector.go:404] processing item [v1/Pod, namespace: gc-9825, name: pod1, uid: e844aac1-d0ff-4e26-973b-efdf2bd7e1fd]
I0213 05:31:03.643952       1 garbagecollector.go:531] adding [v1/Pod, namespace: gc-9825, name: pod2, uid: 0a8973c2-3e63-46c1-bd60-a66952d36c23] to attemptToDelete, because its owner [v1/Pod, namespace: gc-9825, name: pod1, uid: e844aac1-d0ff-4e26-973b-efdf2bd7e1fd] is deletingDependents
I0213 05:31:03.643979       1 garbagecollector.go:404] processing item [v1/Pod, namespace: gc-9825, name: pod2, uid: 0a8973c2-3e63-46c1-bd60-a66952d36c23]
I0213 05:31:04.203429       1 garbagecollector.go:459] object garbagecollector.objectReference{OwnerReference:v1.OwnerReference{APIVersion:"v1", Kind:"Pod", Name:"pod3", UID:"3e113e73-96f7-41aa-a0f0-2c45e53413ed", Controller:(*bool)(0xc002e8ff0a), BlockOwnerDeletion:(*bool)(0xc002e8ff0b)}, Namespace:"gc-9825"} has at least one existing owner: []v1.OwnerReference{v1.OwnerReference{APIVersion:"v1", Kind:"Pod", Name:"pod2", UID:"0a8973c2-3e63-46c1-bd60-a66952d36c23", Controller:(*bool)(0xc0082e3b96), BlockOwnerDeletion:(*bool)(0xc0082e3b97)}}, will not garbage collect
I0213 05:31:04.212855       1 garbagecollector.go:493] at least one owner of object [v1/Pod, namespace: gc-9825, name: pod2, uid: 0a8973c2-3e63-46c1-bd60-a66952d36c23] has FinalizerDeletingDependents, and the object itself has dependents, so it is going to be deleted in Foreground
I0213 05:31:09.499781       1 garbagecollector.go:404] processing item [v1/Pod, namespace: gc-9825, name: pod3, uid: 3e113e73-96f7-41aa-a0f0-2c45e53413ed]
I0213 05:31:09.500342       1 garbagecollector.go:404] processing item [v1/Pod, namespace: gc-9825, name: pod2, uid: 0a8973c2-3e63-46c1-bd60-a66952d36c23]
I0213 05:31:09.907443       1 garbagecollector.go:531] adding [v1/Pod, namespace: gc-9825, name: pod3, uid: 3e113e73-96f7-41aa-a0f0-2c45e53413ed] to attemptToDelete, because its owner [v1/Pod, namespace: gc-9825, name: pod2, uid: 0a8973c2-3e63-46c1-bd60-a66952d36c23] is deletingDependents
I0213 05:31:10.547493       1 garbagecollector.go:482] processing object [v1/Pod, namespace: gc-9825, name: pod3, uid: 3e113e73-96f7-41aa-a0f0-2c45e53413ed], some of its owners and its dependent [[v1/Pod, namespace: gc-9825, name: pod1, uid: e844aac1-d0ff-4e26-973b-efdf2bd7e1fd]] have FinalizerDeletingDependents, to prevent potential cycle, its ownerReferences are going to be modified to be non-blocking, then the object is going to be deleted with Foreground
I0213 05:31:11.221177       1 garbagecollector.go:493] at least one owner of object [v1/Pod, namespace: gc-9825, name: pod3, uid: 3e113e73-96f7-41aa-a0f0-2c45e53413ed] has FinalizerDeletingDependents, and the object itself has dependents, so it is going to be deleted in Foreground
I0213 05:31:16.926296       1 garbagecollector.go:404] processing item [v1/Pod, namespace: gc-9825, name: pod2, uid: 0a8973c2-3e63-46c1-bd60-a66952d36c23]
I0213 05:31:30.274206       1 garbagecollector.go:526] remove DeleteDependents finalizer for item [v1/Pod, namespace: gc-9825, name: pod2, uid: 0a8973c2-3e63-46c1-bd60-a66952d36c23]
I0213 05:31:30.274522       1 garbagecollector.go:404] processing item [v1/Pod, namespace: gc-9825, name: pod3, uid: 3e113e73-96f7-41aa-a0f0-2c45e53413ed]
I0213 05:31:31.473958       1 garbagecollector.go:482] processing object [v1/Pod, namespace: gc-9825, name: pod3, uid: 3e113e73-96f7-41aa-a0f0-2c45e53413ed], some of its owners and its dependent [[v1/Pod, namespace: gc-9825, name: pod1, uid: e844aac1-d0ff-4e26-973b-efdf2bd7e1fd]] have FinalizerDeletingDependents, to prevent potential cycle, its ownerReferences are going to be modified to be non-blocking, then the object is going to be deleted with Foreground
I0213 05:31:32.288039       1 garbagecollector.go:493] at least one owner of object [v1/Pod, namespace: gc-9825, name: pod3, uid: 3e113e73-96f7-41aa-a0f0-2c45e53413ed] has FinalizerDeletingDependents, and the object itself has dependents, so it is going to be deleted in Foreground
I0213 05:31:38.864741       1 garbagecollector.go:404] processing item [v1/Pod, namespace: gc-9825, name: pod1, uid: e844aac1-d0ff-4e26-973b-efdf2bd7e1fd]
I0213 05:31:38.865106       1 garbagecollector.go:404] processing item [v1/Pod, namespace: gc-9825, name: pod3, uid: 3e113e73-96f7-41aa-a0f0-2c45e53413ed]
I0213 05:31:49.848729       1 garbagecollector.go:404] processing item [v1/Pod, namespace: gc-9825, name: pod3, uid: 3e113e73-96f7-41aa-a0f0-2c45e53413ed]
I0213 05:31:49.850981       1 garbagecollector.go:404] processing item [v1/Pod, namespace: gc-9825, name: pod1, uid: e844aac1-d0ff-4e26-973b-efdf2bd7e1fd]
I0213 05:31:57.967514       1 garbagecollector.go:526] remove DeleteDependents finalizer for item [v1/Pod, namespace: gc-9825, name: pod1, uid: e844aac1-d0ff-4e26-973b-efdf2bd7e1fd]
I0213 05:32:27.111824       1 garbagecollector.go:404] processing item [v1/Pod, namespace: gc-9825, name: pod3, uid: 3e113e73-96f7-41aa-a0f0-2c45e53413ed]
I0213 05:32:45.394456       1 garbagecollector.go:526] remove DeleteDependents finalizer for item [v1/Pod, namespace: gc-9825, name: pod3, uid: 3e113e73-96f7-41aa-a0f0-2c45e53413ed]
```

**Comment by apelisse** [comment]
/test pull-kubernetes-e2e-gce

I can't believe it passed!

**Comment by lavalamp** [comment]
This most recent run shows the GC flakes...

**Comment by apelisse** [comment]
Yeah I’ve spent some time looking at it already, not super helpful yet

**Comment by lavalamp** [comment]
in an unrelated PR I found the following in controller manager logs. Logging a ton of extra stuff could cause weird hangups waiting for I/O?

```
I0214 22:04:44.851576       1 stateful_set.go:222] Pod ss2-2 updated, objectMeta {Name:ss2-2 GenerateName:ss2- Namespace:statefulset-242 SelfLink:/api/v1/namespaces/statefulset-242/pods/ss2-2 UID:dad7ba3a-014e-43d5-90da-facb1498df50 ResourceVersion:35432 Generation:0 CreationTimestamp:2020-02-14 22:04:29 +0000 UTC DeletionTimestamp:<nil> DeletionGracePeriodSeconds:<nil> Labels:map[baz:blah controller-revision-hash:ss2-65c7964b94 foo:bar statefulset.kubernetes.io/pod-name:ss2-2] Annotations:map[kubernetes.io/psp:e2e-test-privileged-psp] OwnerReferences:[{APIVersion:apps/v1 Kind:StatefulSet Name:ss2 UID:5b84a6f3-6ddc-42e8-b98d-aa5df8d7782e Controller:0xc003c355a7 BlockOwnerDeletion:0xc003c355a8}] Finalizers:[] ClusterName: ManagedFields:[{Manager:kube-controller-manager Operation:Update APIVersion:v1 Time:2020-02-14 22:04:28 +0000 UTC FieldsType:FieldsV1 FieldsV1:&FieldsV1{Raw:*[123 34 102 58 109 101 116 97 100 97 116 97 34 58 123 34 102 58 103 101 110 101 114 97 116 101 78 97 109 101 34 58 123 125 44 34 102 58 108 97 98 101 108 115 34 58 123 34 46 34 58 123 125 44 34 102 58 98 97 122 34 58 123 125 44 34 102 58 99 111 110 116 114 111 108 108 101 114 45 114 101 118 105 115 105 111 110 45 104 97 115 104 34 58 123 125 44 34 102 58 102 111 111 34 58 123 125 44 34 102 58 115 116 97 116 101 102 117 108 115 101 116 46 107 117 98 101 114 110 101 116 101 115 46 105 111 47 112 111 100 45 110 97 109 101 34 58 123 125 125 44 34 102 58 111 119 110 101 114 82 101 102 101 114 101 110 99 101 115 34 58 123 34 46 34 58 123 125 44 34 107 58 123 92 34 117 105 100 92 34 58 92 34 53 98 56 52 97 54 102 51 45 54 100 100 99 45 52 50 101 56 45 98 57 56 100 45 97 97 53 100 102 56 100 55 55 56 50 101 92 34 125 34 58 123 34 46 34 58 123 125 44 34 102 58 97 112 105 86 101 114 115 105 111 110 34 58 123 125 44 34 102 58 98 108 111 99 107 79 119 110 101 114 68 101 108 101 116 105 111 110 34 58 123 125 44 34 102 58 99 111 110 116 114 111 108 108 101 114 34 58 123 125 44 34 102 58 107 105 110 100 34 58 123 125 44 34 102 58 110 97 109 101 34 58 123 125 44 34 102 58 117 105 100 34 58 123 125 125 125 125 44 34 102 58 115 112 101 99 34 58 123 34 102 58 99 111 110 116 97 105 110 101 114 115 34 58 123 34 107 58 123 92 34 110 97 109 101 92 34 58 92 34 119 101 98 115 101 114 118 101 114 92 34 125 34 58 123 34 46 34 58 123 125 44 34 102 58 105 109 97 103 101 34 58 123 125 44 34 102 58 105 109 97 103 101 80 117 108 108 80 111 108 105 99 121 34 58 123 125 44 34 102 58 110 97 109 101 34 58 123 125 44 34 102 58 114 101 97 100 105 110 101 115 115 80 114 111 98 101 34 58 123 34 46 34 58 123 125 44 34 102 58 102 97 105 108 117 114 101 84 104 114 101 115 104 111 108 100 34 58 123 125 44 34 102 58 104 116 116 112 71 101 116 34 58 123 34 46 34 58 123 125 44 34 102 58 112 97 116 104 34 58 123 125 44 34 102 58 112 111 114 116 34 58 123 125 44 34 102 58 115 99 104 101 109 101 34 58 123 125 125 44 34 102 58 112 101 114 105 111 100 83 101 99 111 110 100 115 34 58 123 125 44 34 102 58 115 117 99 99 101 115 115 84 104 114 101 115 104 111 108 100 34 58 123 125 44 34 102 58 116 105 109 101 111 117 116 83 101 99 111 110 100 115 34 58 123 125 125 44 34 102 58 114 101 115 111 117 114 99 101 115 34 58 123 125 44 34 102 58 116 101 114 109 105 110 97 116 105 111 110 77 101 115 115 97 103 101 80 97 116 104 34 58 123 125 44 34 102 58 116 101 114 109 105 110 97 116 105 111 110 77 101 115 115 97 103 101 80 111 108 105 99 121 34 58 123 125 125 125 44 34 102 58 100 110 115 80 111 108 105 99 121 34 58 123 125 44 34 102 58 101 110 97 98 108 101 83 101 114 118 105 99 101 76 105 110 107 115 34 58 123 125 44 34 102 58 104 111 115 116 110 97 109 101 34 58 123 125 44 34 102 58 114 101 115 116 97 114 116 80 111 108 105 99 121 34 58 123 125 44 34 102 58 115 99 104 101 100 117 108 101 114 78 97 109 101 34 58 123 125 44 34 102 58 115 101 99 117 114 105 116 121 67 111 110 116 101 120 116 34 58 123 125 44 34 102 58 115 117 98 100 111 109 97 105 110 34 58 123 125 44 34 102 58 116 101 114 109 105 110 97 116 105 111 110 71 114 97 99 101 80 101 114 105 111 100 83 101 99 111 110 100 115 34 58 123 125 125 125],}}]} -> {Name:ss2-2 GenerateName:ss2- Namespace:statefulset-242 SelfLink:/api/v1/namespaces/statefulset-242/pods/ss2-2 UID:dad7ba3a-014e-43d5-90da-facb1498df50 ResourceVersion:35543 Generation:0 CreationTimestamp:2020-02-14 22:04:29 +0000 UTC DeletionTimestamp:<nil> DeletionGracePeriodSeconds:<nil> Labels:map[baz:blah controller-revision-hash:ss2-65c7964b94 foo:bar statefulset.kubernetes.io/pod-name:ss2-2] Annotations:map[kubernetes.io/psp:e2e-test-privileged-psp] OwnerReferences:[{APIVersion:apps/v1 Kind:StatefulSet Name:ss2 UID:5b84a6f3-6ddc-42e8-b98d-aa5df8d7782e Controller:0xc0091cf727 BlockOwnerDeletion:0xc0091cf728}] Finalizers:[] ClusterName: ManagedFields:[{Manager:kube-controller-manager Operation:Update APIVersion:v1 Time:2020-02-14 22:04:28 +0000 UTC FieldsType:FieldsV1 FieldsV1:&FieldsV1{Raw:*[123 34 102 58 109 101 116 97 100 97 116 97 34 58 123 34 102 58 103 101 110 101 114 97 116 101 78 97 109 101 34 58 123 125 44 34 102 58 108 97 98 101 108 115 34 58 123 34 46 34 58 123 125 44 34 102 58 98 97 122 34 58 123 125 44 34 102 58 99 111 110 116 114 111 108 108 101 114 45 114 101 118 105 115 105 111 110 45 104 97 115 104 34 58 123 125 44 34 102 58 102 111 111 34 58 123 125 44 34 102 58 115 116 97 116 101 102 117 108 115 101 116 46 107 117 98 101 114 110 101 116 101 115 46 105 111 47 112 111 100 45 110 97 109 101 34 58 123 125 125 44 34 102 58 111 119 110 101 114 82 101 102 101 114 101 110 99 101 115 34 58 123 34 46 34 58 123 125 44 34 107 58 123 92 34 117 105 100 92 34 58 92 34 53 98 56 52 97 54 102 51 45 54 100 100 99 45 52 50 101 56 45 98 57 56 100 45 97 97 53 100 102 56 100 55 55 56 50 101 92 34 125 34 58 123 34 46 34 58 123 125 44 34 102 58 97 112 105 86 101 114 115 105 111 110 34 58 123 125 44 34 102 58 98 108 111 99 107 79 119 110 101 114 68 101 108 101 116 105 111 110 34 58 123 125 44 34 102 58 99 111 110 116 114 111 108 108 101 114 34 58 123 125 44 34 102 58 107 105 110 100 34 58 123 125 44 34 102 58 110 97 109 101 34 58 123 125 44 34 102 58 117 105 100 34 58 123 125 125 125 125 44 34 102 58 115 112 101 99 34 58 123 34 102 58 99 111 110 116 97 105 110 101 114 115 34 58 123 34 107 58 123 92 34 110 97 109 101 92 34 58 92 34 119 101 98 115 101 114 118 101 114 92 34 125 34 58 123 34 46 34 58 123 125 44 34 102 58 105 109 97 103 101 34 58 123 125 44 34 102 58 105 109 97 103 101 80 117 108 108 80 111 108 105 99 121 34 58 123 125 44 34 102 58 110 97 109 101 34 58 123 125 44 34 102 58 114 101 97 100 105 110 101 115 115 80 114 111 98 101 34 58 123 34 46 34 58 123 125 44 34 102 58 102 97 105 108 117 114 101 84 104 114 101 115 104 111 108 100 34 58 123 125 44 34 102 58 104 116 116 112 71 101 116 34 58 123 34 46 34 58 123 125 44 34 102 58 112 97 116 104 34 58 123 125 44 34 102 58 112 111 114 116 34 58 123 125 44 34 102 58 115 99 104 101 109 101 34 58 123 125 125 44 34 102 58 112 101 114 105 111 100 83 101 99 111 110 100 115 34 58 123 125 44 34 102 58 115 117 99 99 101 115 115 84 104 114 101 115 104 111 108 100 34 58 123 125 44 34 102 58 116 105 109 101 111 117 116 83 101 99 111 110 100 115 34 58 123 125 125 44 34 102 58 114 101 115 111 117 114 99 101 115 34 58 123 125 44 34 102 58 116 101 114 109 105 110 97 116 105 111 110 77 101 115 115 97 103 101 80 97 116 104 34 58 123 125 44 34 102 58 116 101 114 109 105 110 97 116 105 111 110 77 101 115 115 97 103 101 80 111 108 105 99 121 34 58 123 125 125 125 44 34 102 58 100 110 115 80 111 108 105 99 121 34 58 123 125 44 34 102 58 101 110 97 98 108 101 83 101 114 118 105 99 101 76 105 110 107 115 34 58 123 125 44 34 102 58 104 111 115 116 110 97 109 101 34 58 123 125 44 34 102 58 114 101 115 116 97 114 116 80 111 108 105 99 121 34 58 123 125 44 34 102 58 115 99 104 101 100 117 108 101 114 78 97 109 101 34 58 123 125 44 34 102 58 115 101 99 117 114 105 116 121 67 111 110 116 101 120 116 34 58 123 125 44 34 102 58 115 117 98 100 111 109 97 105 110 34 58 123 125 44 34 102 58 116 101 114 109 105 110 97 116 105 111 110 71 114 97 99 101 80 101 114 105 111 100 83 101 99 111 110 100 115 34 58 123 125 125 125],}} {Manager:kubelet Operation:Update APIVersion:v1 Time:2020-02-14 22:04:42 +0000 UTC FieldsType:FieldsV1 FieldsV1:&FieldsV1{Raw:*[123 34 102 58 115 116 97 116 117 115 34 58 123 34 102 58 99 111 110 100 105 116 105 111 110 115 34 58 123 34 107 58 123 92 34 116 121 112 101 92 34 58 92 34 67 111 110 116 97 105 110 101 114 115 82 101 97 100 121 92 34 125 34 58 123 34 46 34 58 123 125 44 34 102 58 108 97 115 116 80 114 111 98 101 84 105 109 101 34 58 123 125 44 34 102 58 108 97 115 116 84 114 97 110 115 105 116 105 111 110 84 105 109 101 34 58 123 125 44 34 102 58 109 101 115 115 97 103 101 34 58 123 125 44 34 102 58 114 101 97 115 111 110 34 58 123 125 44 34 102 58 115 116 97 116 117 115 34 58 123 125 44 34 102 58 116 121 112 101 34 58 123 125 125 44 34 107 58 123 92 34 116 121 112 101 92 34 58 92 34 73 110 105 116 105 97 108 105 122 101 100 92 34 125 34 58 123 34 46 34 58 123 125 44 34 102 58 108 97 115 116 80 114 111 98 101 84 105 109 101 34 58 123 125 44 34 102 58 108 97 115 116 84 114 97 110 115 105 116 105 111 110 84 105 109 101 34 58 123 125 44 34 102 58 115 116 97 116 117 115 34 58 123 125 44 34 102 58 116 121 112 101 34 58 123 125 125 44 34 107 58 123 92 34 116 121 112 101 92 34 58 92 34 82 101 97 100 121 92 34 125 34 58 123 34 46 34 58 123 125 44 34 102 58 108 97 115 116 80 114 111 98 101 84 105 109 101 34 58 123 125 44 34 102 58 108 97 115 116 84 114 97 110 115 105 116 105 111 110 84 105 109 101 34 58 123 125 44 34 102 58 109 101 115 115 97 103 101 34 58 123 125 44 34 102 58 114 101 97 115 111 110 34 58 123 125 44 34 102 58 115 116 97 116 117 115 34 58 123 125 44 34 102 58 116 121 112 101 34 58 123 125 125 125 44 34 102 58 99 111 110 116 97 105 110 101 114 83 116 97 116 117 115 101 115 34 58 123 125 44 34 102 58 104 111 115 116 73 80 34 58 123 125 44 34 102 58 115 116 97 114 116 84 105 109 101 34 58 123 125 125 125],}}]}.
```

**Comment by liggitt** [comment]
but that's useful ... you'd want to know if it was `116 97 117 101 102 117 108 115` instead of `116 97 116 101 102 117 108 115`

**Comment by apelisse** [comment]
I've changed it to: `ManagedFields:[{Manager:kube-controller-manager Operation:Update APIVersion:v1 Time:2020-02-19 22:15:03 +0000 UTC FieldsType:FieldsV1 Fields\
V1:[snipped...]} {Manager:kube-scheduler Operation:Update APIVersion:v1 Time:2020-02-19 22:15:03 +0000 UTC FieldsType:FieldsV1 FieldsV1:[snipped...]}]`

**Comment by apelisse** [comment]
Garbage collection didn't fail ...
/test pull-kubernetes-e2e-gce

**Comment by apelisse** [comment]
Garbage collection didn't fail ...
/test pull-kubernetes-e2e-gce

**Comment by apelisse** [comment]
gce test passed, unbelievable :-)
/test pull-kubernetes-e2e-gce

**Comment by apelisse** [comment]
gce test passed, unbelievable :-)
/test pull-kubernetes-e2e-gce

**Comment by apelisse** [comment]
Garbage collection didn't fail ...
/test pull-kubernetes-e2e-gce

**Comment by jpbetz** [comment]
I compared the controller manager memory pprofile of `pull-kubernetes-e2e-gce-100-performance` for this PR vs https://github.com/kubernetes/kubernetes/pull/88297 (which also has SSA at 100%). In both cases, `FieldsV1.Unmarshal` is in the top 3 for memory allocation (by bytes). With this PR, it drops from 2nd to third place, which suggests this is an improvement, but this is not very precise.

(https://gist.github.com/jpbetz/81331fb13fbbe4336fa62343e853180c, https://gist.github.com/jpbetz/0ee66a5712fd280cf0729ce46f0189b4)

So removing managers from metadataonly might help a bit (3%?), removing it more broadly might help more, the upper bound is a ~10% bytes allocated improvement. The GC pressure that creates might reduce latency as well. I don't know by how much. I don't see any direct impact controller manager cpu by SSA, so I suspect it's primarily due to the memory allocated + increased resource utilization on the other master components due to enabling SSA.

Based on the GC e2e failures, the latency added by SSA was in the 20-35% range. So if we might get maybe 5-10% of that back here if we managers broadly enough.

**Comment by jpbetz** [comment]
`FieldsV1.Unmarshal` is also in the top 3 for memory allocations for the api-server, accounting for ~5.5% of bytes allocated. Getting some of that back would be valuable. I don't see SSA otherwise contributing to api-server latency significantly in the pprof data.

**Comment by apelisse** [comment]
Experiment is over.
/close

**Pull request closed without merging**